File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Source/Tva.Phasors/Ieee1344/ICommonFrameHeader.vb
'*******************************************************************************************************
'  ICommonFrameHeader.vb - IEEE 1344 Common frame header interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: James R Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  01/14/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Imports Tva.DateTime

Namespace Ieee1344

    <CLSCompliant(False)> _
    Public Interface ICommonFrameHeader

        Inherits IChannelFrame

        Shadows Property IDCode() As UInt64

        Shadows ReadOnly Property FrameType() As FrameType

        ReadOnly Property FrameLength() As Int16

        ReadOnly Property DataLength() As Int16

        Shadows ReadOnly Property TimeTag() As NtpTimeTag

        Property SynchronizationIsValid() As Boolean

        Property DataIsValid() As Boolean

        Property TriggerStatus() As TriggerStatus

        Property InternalSampleCount() As Int16

        Property InternalStatusFlags() As Int16

    End Interface

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/Ieee1344/ICommonFrameHeader.cs
//*******************************************************************************************************
//  IFrameImage.cs
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;
using PCS.Parsing;

namespace PCS.PhasorProtocols.Ieee1344
{
    /// <summary>
    /// Represents the interface that uniquely identifies IEEE 1344 frame images (see <see cref="FrameImageParserBase{TTypeIdentifier,TOutputType}"/>).
    /// </summary>
    [CLSCompliant(false)]
    public interface IFrameImage : ISupportFrameImage<FrameType>, IChannelFrame
    {
    }
}